.github/workflows/run-linters.yaml

```diff
@@ -0,0 +1,17 @@
+---
+name: Pre-commit
+
+on:
+  pull_request:
+    branches:
+      - main
+
+  workflow_dispatch:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
I think we should use v4 here right ? That's the latest.

re, v3: https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/

Reply:
Thank you! Changed to `v4`